source/docs/flow_control.md

```diff
@@ -114,9 +114,95 @@ The low watermark path is similar
  `ConnectionImpl::StreamImpl::pendingRecvBufferLowWatermark`.
  * `pendingRecvBufferLowWatermarkwhich` calls `readDisable(false)` on the stream.
 
-## HTTP/2 filters
+## HTTP/1 and HTTP/2 filters
+
+Each HTTP and HTTP/2 filter has an opportunity to call `decoderBufferLimit()` or
+`encoderBufferLimit()` on creation.  No filter should buffer more than the
+configured bytes without calling the appropriate watermark callbacks or sending
+an error response.
+
+Filters may override the default limit with calls to `setDecoderBufferLimit()`
+and `setEncoderBufferLimit()`.  These calls should only increase the buffer
+limit, and will be ignored if they would reduce the limit.
+
+Most filters do not buffer internally, but instead push back on data by
+returning a FilterDataStatus on `encodeData()`/`decodeData()` calls.
+If a buffer is a streaming buffer, i.e. the buffered data will resolve over
+time, it should return `FilterDataStatus::StopIterationAndWatermark` to pause
+further data processing, which will cause the `ConnectionManagerImpl` to trigger
+watermark callbacks on behalf of the filter.  If a filter can not make forward progress without the
+complete body, it should return `FilterDataStatus::StopIterationAndWatermark`.
+in this case if the `ConnectionManagerImpl` buffers more than the allowed data
+it will return an error downstream: a 413 on the request path, 500 or `resetStream()` on the
+response path.
+
+# Decoder filters
```

Comment:
Do this sections need to be updated now to indicate that connection manager is doing most/all of the work?

Reply:
So the block above talks about filters in general using the connection manager, then the encoder/decoder state 'most don't buffer internally but for those that do'
If you can think of a way to add more clarity let me know :-)